### PR TITLE
enable use_key_in_widget_constructors lint

### PR DESCRIPTION
**Pull request by goderbauer**
https://github.com/flutter/flutter/issues/75663

### CODE REVIEW

File: analysis_options.yaml [hunk 1]
```diff
@@ -168,7 +168,7 @@ linter:
     - prefer_initializing_formals
     - prefer_inlined_adds
     # - prefer_int_literals # not yet tested
```
Comment (Hixie) [inline analysis_options.yaml:170]: while you're at it you could say that `prefer_int_literals # conflicts with https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#use-double-literals-for-double-constants`

File: dev/benchmarks/complex_layout/lib/main.dart [hunk 2]
```diff
@@ -7,13 +7,15 @@ import 'package:flutter/scheduler.dart' show timeDilation;
```
Comment (Hixie) [inline dev/benchmarks/complex_layout/lib/main.dart:7]: do these changes affect benchmark numbers?

**Review by Hixie — APPROVED** [review]
LGTM

File: dev/benchmarks/complex_layout/lib/main.dart (hunk 2 above)
Comment (goderbauer) [reply dev/benchmarks/complex_layout/lib/main.dart:7]: Will keep an eye on the numbers.

**Pull request merged**